tests/agent_server/test_agent_code_loading.py

```diff
@@ -322,3 +324,98 @@ def test():
+        version = await clienthelper.get_version()
+        res = await client.put_version(
+            tid=environment,
+            version=version,
+            resources=[],
+            pip_config=PipConfig(),
+            compiler_version=get_compiler_version(),
+        )
+        assert res.code == 200
+        return version
+
+    version_1 = await get_version()
+
+    res = await client.upload_code_batched(tid=environment, id=version_1, resources={"test::Test": sources})
+    assert res.code == 200
+
+    res = await client.upload_code_batched(tid=environment, id=version_1, resources={"test::Test2": sources})
+    assert res.code == 200
+
+    res = await client.upload_code_batched(tid=environment, id=version_1, resources={"test::Test3": sources})
+    assert res.code == 200
+
+    agent: Agent = await agent_factory(
+        environment=environment, agent_map={"agent1": "localhost"}, hostname="host", agent_names=["agent1"], code_loader=True
+    )
+
+    resource_install_specs_1: list[ResourceInstallSpec]
+    resource_install_specs_2: list[ResourceInstallSpec]
+
+    # We want to test
+    resource_install_specs_1, invalid_resource_types_1 = await agent.get_code(
+        environment=environment, version=-1, resource_types=["test::Test", "test::Test2", "test::Test3"]
+    )
+    assert len(invalid_resource_types_1) == 3
+    for invalid_resource in invalid_resource_types_1:
+        assert (
+            "Failed to get source code for " + invalid_resource.resource_type + " version=-1, result={'message': 'Request or "
+            "referenced resource does not exist: The version of the code does not exist. "
+            + invalid_resource.resource_type
+            + ", -1'}"
+        ) == str(invalid_resource.exception)
+
+    await agent.ensure_code(
+        code=resource_install_specs_1,
+    )
+
+    resource_install_specs_2, _ = await agent.get_code(
+        environment=environment, version=version_1, resource_types=["test::Test", "test::Test2"]
+    )
+
+    async def _install(blueprint: executor.ExecutorBlueprint) -> None:
+        raise Exception("MKPTCH: Unable to load code when agent is started with code loading disabled.")
+
+    monkeypatch.setattr(agent, "_install", _install)
+
+    failed_to_load = await agent.ensure_code(
+        code=resource_install_specs_2,
+    )
+    assert len(failed_to_load) == 2
+    for failed_resource in failed_to_load:
+        assert "MKPTCH: Unable to load code when agent is started with code loading disabled" in str(failed_resource.exception)
+
+    monkeypatch.undo()
+
+    idx1 = log_index(
+        caplog,
+        "inmanta.agent.agent",
+        logging.ERROR,
+        "Failed to get source code for test::Test2 version=-1",
+    )
+
+    log_index(caplog, "inmanta.agent.agent", logging.ERROR, "Failed to install handler test::Test version=1", idx1)
+
+    log_index(caplog, "inmanta.agent.agent", logging.ERROR, "Failed to install handler test::Test2 version=1", idx1)
```

Comment:
It would be interesting to also make assertions on the resource action log

Reply:
These assertions have been added to the other test